sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -672,7 +672,7 @@ def send_batch(self, event_data_batch, **kwargs):
                 )
                 if self._on_success:
                     self._on_success(batch._internal_events, pid)
-            except (KeyError, AttributeError, EventHubError) as e:
+            except (KeyError, AttributeError, EventHubError) as e:  # TODO: do we want all EventHubErrors to be caught?
```

Comment:
@kashifkhan - do we want all EventHubErrors to be caught? b/c, for ex, OperationSendTimeout gets caught and re-run here, even though it can be raised. (It'll be raised anyway when the except code is run again, but it adds an extra run with the same outcome so just checking)

Reply:
it would be raised again if the send operation times out again, but for example if whatever cause the timeout is now gone and you can have a successful send and we decide to raise we  could lose out on the opportunity for a send?

will it hit the retry if its raised instead of going through the except block?